src/lib.rs

```diff
@@ -100,6 +104,60 @@
 //! (`let arr: Array3<f32>`) or via a turbofish operator (`read_geotiff::<f32, _>`).
 //! Currently supported dtypes include uint (u8/u16/u32/u64), int (i8/i16/i32/i64) and
 //! float (f16/f32/f64).
+//!
+//! # GPU (CUDA) decoder
+//!
+//! Uses [`nvtiff_sys`] to decode TIFF images. Pixel data is stored as a flattened 1D
+//! array in row-major order (i.e. rows-first, columns-next). Use:
+//! - [`CudaCogReader`](crate::io::nvtiff::CudaCogReader) - returns a
+//!   [DLPack](https://dmlc.github.io/dlpack/latest) in-memory data structure via
+//!   [`dlpark`](https://docs.rs/dlpark)
+//!
+//! ## Examples
+//!
+//! ### DLPack
+//!
+//! Retrieve a GeoTIFF file stream via the [`object_store`] crate, and set up a CUDA
+//! stream via the [`cudarc`] crate. Pass the file stream and CUDA stream into the
+//! [`CudaCogReader::new`](crate::io::nvtiff::CudaCogReader::new) method to instantiate
+//! a [`CudaCogReader`](crate::io::nvtiff::CudaCogReader) struct, and call
+//! [`.dlpack()`](crate::io::nvtiff::CudaCogReader::dlpack) to get a
+//! [`dlpark::SafeManagedTensorVersioned`] output.
+//!
+//! ```rust
+//! use std::sync::Arc;
+//!
+//! use bytes::Bytes;
+//! use cog3pio::io::nvtiff::CudaCogReader;
+//! use cudarc::driver::{CudaContext, CudaStream};
+//! use dlpark::SafeManagedTensorVersioned;
+//! use dlpark::ffi::DataType;
+//! use dlpark::prelude::TensorView;
+//! use object_store::path::Path;
+//! use object_store::{GetResult, ObjectStore, parse_url};
+//! use tokio;
+//! use url::Url;
+//!
+//! #[tokio::main]
+//! async fn main() {
+//!     let cog_url: &str =
+//!         "https://github.com/cogeotiff/rio-tiler/raw/7.9.0/tests/fixtures/cog_nodata_float_nan.tif";
+//!     let tif_url: Url = Url::parse(cog_url).unwrap();
+//!     let (store, location): (Box<dyn ObjectStore>, Path) = parse_url(&tif_url).unwrap();
+//!
+//!     let result: GetResult = store.get(&location).await.unwrap();
+//!     let bytes: Bytes = result.bytes().await.unwrap();
+//!
+//!     let ctx: Arc<CudaContext> = cudarc::driver::CudaContext::new(0).unwrap(); // Set on GPU:0
+//!     let cuda_stream: Arc<CudaStream> = ctx.default_stream();
+//!
+//!     // Read GeoTIFF into a dlpark::versioned::SafeManagedTensorVersioned
+//!     let mut cog = CudaCogReader::new(&bytes, &cuda_stream).unwrap();
+//!     let tensor: SafeManagedTensorVersioned = cog.dlpack().unwrap();
+//!     assert_eq!(tensor.shape(), [7088886]); // [1, 2667, 2658]
+//!     assert_eq!(tensor.data_type(), &DataType::F32);
+//! }
+//! ```
```

Comment:
This doctest fails on CI with "error[E0433]: failed to resolve: use of unresolved module or unlinked crate `cudarc`". Need to find a way to hide this the "cuda" feature flag also, and maybe use something like https://github.com/dtolnay/cargo-docs-rs to ensure that builds still work on docs.rs later.